### PR TITLE
19w38b - loot conditions from data packs, advancements go smart, data…

### PR DESCRIPTION
**Pull request by liach**
… storage, etc

Also renamed property containers and state factory to state and state managers. when you have something that has configurable properties, you can reliably call it a state; and the state factory is not a factory; it just create all state instances and poll the one you need on demand, which is more like a manager.

Both of these names come from when asie was working on yarn on his own almost one year ago. During this period, no one has considered a better name for these classes and I figured that such shorter names may reduce confusion (especially when you have a StateFactory.Factory class, which is cursed)

The new "predicates" data type is named loot condition manager, as that is actually loot conditions instead of predicates shared by advancements and loot table.

Also consider burning the two classes in `net/minecraft/predicate/block` in a fire as they are in no way resembling the json predicate system in vanilla; they are used for block pattern matching (the block state predicate in that package) and world generation (the block one)

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by LemmaEOF — APPROVED** [review]
I think this is good to merge.

**Review by Gegy — APPROVED** [review]

File: mappings/net/minecraft/state/property/Property.mapping
Comment (Juuxel) [inline mappings/net/minecraft/state/property/Property.mapping:6]: Why `name`? I think `getName` sounds better (for example `Properties.FACING.getName(Direction.UP)`).

Comment (liach) [reply mappings/net/minecraft/state/property/Property.mapping:6]: It is naming the value, hence just the verb `name`. It would be confusing with the method that gets the name for property itself, e.g. `Properties.FACING.getName()` may return `"facing"` while `Properties.FACING.getName(Direction.UP)` may return `"up"`

Comment (Juuxel) [reply mappings/net/minecraft/state/property/Property.mapping:6]: Oh, I didn't see that there was another `getName()`.

Comment (liach) [reply mappings/net/minecraft/state/property/Property.mapping:6]: Yeah, that is what we have now. 
image

File: mappings/net/minecraft/command/StorageDataObject.mapping
Comment (Juuxel) [inline mappings/net/minecraft/command/StorageDataObject.mapping]: `SUGGESTION_PROVIDER` to match the class name?

File: mappings/net/minecraft/predicate/PlayerPredicate.mapping
Comment (Juuxel) [inline mappings/net/minecraft/predicate/PlayerPredicate.mapping:25]: `gameMode` (this capitalization is used in the class name and fields in other classes)

File: mappings/net/minecraft/state/property/Property.mapping
Comment (Juuxel) [reply mappings/net/minecraft/state/property/Property.mapping:6]: In that case something like `getValueName` would sound better. It's not naming the value, it's getting the value's existing name.

Also applies to method_16551.

Comment (liach) [reply mappings/net/minecraft/state/property/Property.mapping:6]: Nah, if you have two properties of the same type, they can actually have different names for values as the name-value mapping is a part of the property.

File: mappings/net/minecraft/predicate/PlayerPredicate.mapping
Comment (liach) [reply mappings/net/minecraft/predicate/PlayerPredicate.mapping:25]: https://minecraft.gamepedia.com/Java_Edition_19w38a#General
It's called `gamemode` in JSON key, so `gamemode` is fine for field name

Comment (Gegy) [reply mappings/net/minecraft/predicate/PlayerPredicate.mapping:25]: Imo should have a standard still -- maybe can rename other fields to `gamemode` like this, or alternatively change this name?

File: mappings/net/minecraft/command/StorageDataObject.mapping
Comment (liach) [reply mappings/net/minecraft/command/StorageDataObject.mapping]: Done

**Comment by liach** [comment]
Asking for some approval and reviews before Wednesday the snapshot day.

**Comment by modmuss50** [comment]
Conflict (im on mobile so cannot merge it) after that it looks good to merge.

**Pull request merged**
